src/quicklist.c

```diff
@@ -2687,30 +2691,40 @@ int quicklistTest(int argc, char *argv[]) {
                         quicklistPushHead(ql, genstr("hello HEAD", i + 1), 64);
                     }
 
-                    quicklistNode *node = ql->head;
-                    unsigned int low_raw = ql->compress;
-                    unsigned int high_raw = ql->len - ql->compress;
-
-                    for (unsigned int at = 0; at < ql->len;
-                         at++, node = node->next) {
-                        if (at < low_raw || at >= high_raw) {
-                            if (node->encoding != QUICKLIST_NODE_ENCODING_RAW) {
-                                ERR("Incorrect compression: node %d is "
-                                    "compressed at depth %d ((%u, %u); total "
-                                    "nodes: %lu; size: %u)",
-                                    at, depth, low_raw, high_raw, ql->len,
-                                    node->sz);
+                    for (int step = 0; step < 2; step++) {
+                        /* test remove node */
+                        if (step == 1) {
+                            for (int i = 0; i < list_sizes[list] / 2; i++) {
```

Comment:
if `list_sizes[list]` is odd I think this won't remove a whole node. Maybe just loop over `q1` popping from head and tail until `q1->len` size decreases.

Reply:
I remove half elements. In most cases the nodes count > 1,  so I think it's OK.
What I want is `ql->len == ql->compress * 2 + 1`, then delete a node. This will make old code fail.

Reply:
Ok, I understand. I thought `list_sizes[list]` is the size of a node, but I misread the code.
But I think you remove either all the elements or all the elements except 1 (and not half), this is because you remove 2 elements in each iteration.

Reply:
Ok, I understand. I thought list_sizes[list] is the size of a node, but I misread the code.
But I think you remove either all the elements or all the elements except 1 (and not half), this is because you remove 2 elements in each iteration.

Reply:
Ok, I understand. I thought `list_sizes[list]` is the size of a node, but I misread the code.
But I think you remove either all the elements or all the elements except 1 (and not half), this is because you remove 2 elements in each iteration.

Reply:
Sorry my bad again. I see `ql` is populated with *2 the elements specified in `list_sizes`. Bad naming, but your code is correct.